service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java

```diff
@@ -580,4 +593,107 @@ public void deleteSiteImage(JWTData userData, int imageId) {
 
     db.deleteFrom(SITE_IMAGES).where(SITE_IMAGES.ID.eq(imageId)).execute();
   }
+
+  @Override
+  public List<FilterSitesResponse> filterSites(JWTData userData, FilterSitesRequest filterSitesRequest) {
+    assertAdminOrSuperAdmin(userData.getPrivilegeLevel());
+
+    // first find all valid siteIds and filter them down
+    Map<Integer, Record3<Integer, String, Integer>> sitesRecords =
```

Comment:
💭 Do you think instead of getting just the site IDs (and their addresses and neighborhood IDs) here, we could [join](https://stackoverflow.com/questions/38385248/how-to-join-3-tables-and-iterate-results-using-jooq) ([official docs](https://www.jooq.org/doc/latest/manual/sql-building/sql-statements/select-statement/join-clause/)) the relevant tables (`ADOPTED_SITES`, `SITES`, `USERS`, and `SITE_ENTRIES` -- we could also try to fit `STEWARDSHIP` here but it could get a little tricky) and filter in one query? The `WHERE` clause is gonna be (a little more than) a little big but we should be able to generate it like it's shown [here](https://stackoverflow.com/questions/37238418/how-to-write-optional-where-clauses-in-jooq) to make it more readable